### PR TITLE
Create `TypedOp` classes for more flexibility.

### PR DESCRIPTION
**Pull request by eriknw**
This began by wanting to use the numba-compiled UDF for scalar `isclose`.
This resulted in paying down technical debt.
We no longer rely on weird global state, and some operations are more specific.

### CODE REVIEW

File: grblas/ops.py [hunk 1]
```diff
@@ -137,31 +200,29 @@ class OpBase:
...
-        self._specific_types[type_] = obj
+    def _add(self, op):
+        self._typed_ops[op.type] = op
+        self.types[op.type] = op.return_type
 
     def __delitem__(self, type_):
-        type_ = self._normalize_type(type_)
-        del self._specific_types[type_]
+        type_ = _normalize_type(type_)
+        del self._typed_ops[type_]
```
Comment (jim22k) [inline grblas/ops.py:221]: Should `type_` also be removed from `self.types` here?

File: grblas/ops.py [hunk 2]
```diff
@@ -590,39 +642,31 @@ def register_new(cls, name, monoid, binaryop):
...
-    return _return_type[gb_op]
+        opclass = gb_op.__class__.__name__
+    elif isinstance(gb_op, TypedOpBase):
+        opclass = gb_op.opclass
+    elif isinstance(gb_op, ParameterizedUdf):
+        gb_op = gb_op()  # Use default parameters of parameterized UDFs
+        gb_op, opclass = find_opclass(gb_op)
+    else:
+        opclass = UNKNOWN_OPCLASS
+    return gb_op, opclass
```
Comment (jim22k) [inline grblas/ops.py:669]: Is this function used anywhere other than in the tests? If not, we should remove it and refactor the tests to use `get_typed_op` instead.

**Comment by jim22k** [comment]
Overall, I like this change. Definitely helps unify various concepts floating around in different functions.

File: grblas/ops.py (hunk 1 above)
Comment (eriknw) [reply grblas/ops.py:221]: Good catch!

File: grblas/ops.py (hunk 2 above)
Comment (eriknw) [reply grblas/ops.py:669]: It's used one place: `grblas.base:GbContainer.__call__` to determine whether an argument in `*optional_mask_and_accum` is a binary op for accum.  I was hoping to remove `find_opclass` as well.

**Comment by eriknw** [comment]
Thanks for looking at this.

Note that I changed the default op for `Matrix.reduce_rows` (and columns) to `binary.add`.  It no longer uses a different default for bool dtypes.  I prefer this for three reasons:
  1. I like the example in the signature so one clearly sees a binary op
  2. I don't want to guess what is means by `A.reduce_rows()`--does it count the true, use `lor`, or use `land`?
  3. It follows all other default operators such as `ewise_mult`, which don't have per-type defaults.

**Comment by eriknw** [comment]
I'm glad you like this overall.  Please don't be shy about sharing small nits.  I've been changing a lot of things that didn't necessarily need changed to better match my sensibilities, but it's okay if you prefer things differently.

**Pull request merged**